src/main/java/io/openliberty/tools/intellij/util/ExceptionUtil.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ ******************************************************************************/
+package io.openliberty.tools.intellij.util;
+
+import com.intellij.openapi.progress.ProcessCanceledException;
+import com.intellij.openapi.project.IndexNotReadyException;
+
+import java.util.concurrent.CancellationException;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import io.openliberty.tools.intellij.lsp4mp4ij.psi.core.java.codeaction.JavaCodeActionResolveContext;
+import io.openliberty.tools.intellij.lsp4mp4ij.psi.core.java.corrections.proposal.ChangeCorrectionProposal;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.WorkspaceEdit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class is intended for cases where exception handling is repetitive
+ */
+public class ExceptionUtil {
+
+    /**
+     * Executes a supplied action with structured exception handling, allowing
+     * the caller to specify a fallback function in case of an exception.
+     *
+     * @param action  the main operation to execute, represented by a Supplier
+     * @param fallback a function to handle exceptions and provide a safe return value
+     * @param <T>     the type of result expected from the action
+     * @return        the result of the action or the fallback value in case of an exception
+     */
+    public static <T> T executeWithExceptionHandling(Supplier<T> action, Function<Exception, T> fallback) {
+        try {
+            return action.get();
+        } catch (ProcessCanceledException e) {
+            //Since 2024.2 ProcessCanceledException extends CancellationException so we can't use multi-catch to keep backward compatibility
+            //TODO delete block when minimum required version is 2024.2
+            throw e;
+        } catch (IndexNotReadyException | CancellationException e) {
+            throw e;
+        } catch (Exception e) {
+            // Invoke the fallback function with the exception to generate a safe return value.
+            return fallback.apply(e);
+        }
+    }
```

Comment:
```suggestion
    public static <T> T executeWithExceptionHandling(Supplier<T> action, Function<RuntimeException, T> fallback) {
        try {
            return action.get();
        } catch (ProcessCanceledException e) {
            //Since 2024.2 ProcessCanceledException extends CancellationException so we can't use multi-catch to keep backward compatibility
            //TODO delete block when minimum required version is 2024.2
            throw e;
        } catch (IndexNotReadyException | CancellationException e) {
            throw e;
        } catch (RuntimeException e) {
            // Invoke the fallback function with the exception to generate a safe return value.
            return fallback.apply(e);
        }
    }
```

Reply:
@vaisakhkannan As we discussed today, here's the suggested change to allow the fallback function to be able to re-throw exceptions.

Reply:
@mrglavas , I just saw your commit suggestion after I pushed the changes I made locally, so I didn’t receive the email at the same time. Could you please check the changes I pushed now?

Reply:
@mrglavas , I made the change